odbc/src/api/bind_col_tests.rs

```diff
@@ -129,6 +129,73 @@ mod unit {
         }
     }
 
+    #[test]
+    fn test_binding_column_with_odbc_2_type_that_is_deprecated_in_odbc_3() {
+        // Set up MongoHandle
```

Comment:
The linked [help ticket](https://jira.mongodb.org/browse/HELP-69793) on this ticket was about how column binding with an ODBC 2.x type was not working, so I made this unit test to show that the SQLBindCol `Invalid target type 4` error is no longer happening.

Reply:
Nice.  Also add to the test to show that it will work for the newly added types as well for `-6` and `5`.

Reply:
I didn't test for those because we don't use their equivalents, `SQL_C_STINYINT` and `SQL_C_SSHORT`, in the ODBC driver. They just exist in our `CDataType` enum.